WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

```diff
@@ -491,7 +493,10 @@ class MySiteViewModel @Inject constructor(
                 )
         )
 
-        val jetpackBadge = JetpackBadge.takeUnless { buildConfigWrapper.isJetpackApp }
+        val jetpackBadge = JetpackBadge.takeUnless {
+            buildConfigWrapper.isJetpackApp
+            !jetpackPoweredFeatureConfig.isEnabled()
```

Comment:
My understanding is that we need an `and` operator in the predicate, otherwise only the last statement is taken into account as per function blocks in Kotlin where the last statement is in fact the `return` value.

I'm also thinking since we're extending the logic here, using `takeIf { /* predicate */ }` could make it easier to read and reason about:

```kotlin
val jetpackBadge = JetpackBadge.takeIf {
    jetpackPoweredFeatureConfig.isEnabled() && !buildConfigWrapper.isJetpackApp
}
```

I'll open a suggestion PR targeting this one, since I have a few updates to the unit tests as well 👍